### ISSUE TITLE
Unnamed databases can be created

### ISSUE THREAD

**Issue by tlvince** [pouchdb/pouchdb#5483]
Currently, it seems omitting the `name` param (or omitting it in `options`) to the `PouchDB` constructor is permitted. Is this expected behaviour?

You can test this with simply `new PouchDB()` or even any falsy value like `new PouchDB(undefined)`; we were bitten by this via `new PouchDB(NaN)` (don't ask 😄), which caused some methods (specifically `replicate.{to,from}`) to be missing from the instance.

PouchDB v5.4.5.

**Comment by daleharvey** [comment]
Surprised since we do name validation, however you are right and null checks are missing, I would expect this to throw an error

**Comment by daleharvey** [comment]
This should be a relatively straightfoward fix for someone who wanted to contribute to PouchDB. 

 * First you will need to clone PouchDB, install CouchDB and get the test suite passing
(https://github.com/pouchdb/pouchdb/blob/master/TESTING.md)

 * With a passing test suite, we should write new tests in https://github.com/pouchdb/pouchdb/blob/master/tests/integration/test.basics.js, something like 

```
try {
  new PouchDB();
} catch (err) { 
  err.should.exist();
  done();
}
```

  * Run the tests, ensure they fail
  * Write the fix, the code already does a string check @ https://github.com/pouchdb/pouchdb/blob/master/packages/pouchdb-core/src/constructor.js#L63 so this should go in the same place (investigate why the string check is not currently firing)

  * Run the tests and ensure they are fixed then open a PR :)

If anyone wants to take this on I am happy to help out with any questions, there is some additional guidance @ https://github.com/pouchdb/pouchdb/blob/master/CONTRIBUTING.md#guide-to-contributions

**Comment by adamlofts** [comment]
I've taken a look at this issue and I think it is already fixed on the master branch. The fix and associated test was added in ebbadb7dd7

**Comment by daleharvey** [comment]
Heh I totally forgot about doing that, sorry about that, I did think it was confusing looking at the code why this wasnt caught.

Cheers, fixed in https://github.com/pouchdb/pouchdb/commit/ebbadb7dd7c88c032cc727f405276ba8cc60f48d
